lib/matplotlib/axes/_axes.py

```diff
@@ -6054,25 +6054,30 @@ def _normalize_input(inp, ename='input'):
             """
             if (isinstance(x, np.ndarray) or
                     not iterable(cbook.safe_first_element(inp))):
-                # TODO: support masked arrays;
-                inp = np.asarray(inp)
+
+                inp = cbook.safe_masked_invalid(inp)
                 if inp.ndim == 2:
                     # 2-D input with columns as datasets; switch to rows
                     inp = inp.T
+
+                    if inp.shape[1] < inp.shape[0]:
```

Comment:
This is not actually just migrated up from where it was before (old line 6068).  I don't care for it either but left it alone thinking it probably got debated when it was added.  I'm more than happy to remove.